Added sort by rank/suit buttons

MeirGavish opened the pull request:
Also used the palette encoding from #350 for the round background so I don't need to change the palette indices in the code.

I thought it could be nice to add these buttons because Balatro has them.
I didn't remove the sort toggle from the R button, wondering if I should. Maybe we can reassign R and L and make them discard hand and play hand quick buttons.

As for code, I made the minimal changes needed to make this work and didn't do any larger refactor to the sorting code, maybe that should be left for #268.

https://github.com/user-attachments/assets/101102f3-3603-48d4-bc4c-47a94b6f086d

MathisMartin31 commented:
That's a nifty feature there ^^

I only have a small nitpick: shouldn't the right button say "SUIT" instead of "SUITE"?

MeirGavish commented on source/game.c:
```diff
@@ -1989,10 +2045,25 @@ static void game_playing_play_hand_on_pressed(void)
+
+    hand_state = HAND_PLAY;
+    display_hands(--hands);
+
```
```suggestion
```

Another merge error, should be removed

MeirGavish commented:
> That's a nifty feature there ^^
> 
> I only have a small nitpick: shouldn't the right button say "SUIT" instead of "SUITE"?

Damn, I can't believe I missed this, thanks!

MeirGavish commented:
> > That's a nifty feature there ^^
> > I only have a small nitpick: shouldn't the right button say "SUIT" instead of "SUITE"?
> 
> Damn, I can't believe I missed this, thanks!

Fixed

<img width="480" height="320" alt="image" src="https://github.com/user-attachments/assets/c12b9389-886b-4103-bc42-ccf95a562160" />

I made the suit button 1 pixel narrower so the text can be centered in it, I'm trying to figure out whether it looks noticeably bad.

MeirGavish commented:
> > > > That's a nifty feature there ^^
> > > > I only have a small nitpick: shouldn't the right button say "SUIT" instead of "SUITE"?
> > > 
> > > 
> > > Damn, I can't believe I missed this, thanks!
> > 
> > 
> > Fixed
> > <img alt="image" width="480" height="320" src="https://private-user-images.githubusercontent.com/34395426/537234725-c12b9389-886b-4103-bc42-ccf95a562160.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njg3NDg1MzgsIm5iZiI6MTc2ODc0ODIzOCwicGF0aCI6Ii8zNDM5NTQyNi81MzcyMzQ3MjUtYzEyYjkzODktODg2Yi00MTAzLWJjNDItY2NmOTVhNTYyMTYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTE4VDE0NTcxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM5ZjhlY2E1OTY4MmE4MzU1ODUyZDRlZjQ1ZTEwYzNiNTdlODU5ODI1Y2NkNWZhODBkYzI0ZGZhMTZjNTU5ODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.n6FXoV10j5GO4Ve_z6cPFcwBf-t60D9M6_mM5zKDMC8">
> > I made the suit button 1 pixel narrower so the text can be centered in it, I'm trying to figure out whether it looks noticeably bad.
> 
> It's kind of noticeable yeah, but I think you can make the S 1px wider to compensate?
> 
> We could also do this for the Discard button, I'm noticing now the text isn't centered either.
> 


Okay, but I can't change just this "S", the font in the UI has to be consistent so I changed all of the 2x5 S's in the UI to 3x5 including "**S**KIP", "**S**ELECT", "**S**MALL BLIND", "BO**SS** BLIND", "DI**S**CARD".
I took the S from the [m3x6](https://managore.itch.io/m3x6) font which is the closest parallel to Balatro's [m6x11](https://managore.itch.io/m6x11) font at this size. We should probably make a standardized font for the UI based on it.
The thing is, now the "SELECT" text during blind select is not centered within the button. The SKIP button is centered now though... Maybe centering all text in the buttons is a little much to hope for
<img width="480" height="320" alt="image" src="https://github.com/user-attachments/assets/27051898-ec83-4343-82a8-1add47066b73" />
<img width="480" height="320" alt="image" src="https://github.com/user-attachments/assets/a097d4e4-c0e7-4d62-82de-607677c552d1" />

MeirGavish commented:
> > > > > > That's a nifty feature there ^^
> > > > > > I only have a small nitpick: shouldn't the right button say "SUIT" instead of "SUITE"?
> > > > > 
> > > > > 
> > > > > Damn, I can't believe I missed this, thanks!
> > > > 
> > > > 
> > > > Fixed
> > > > <img alt="image" width="480" height="320" src="https://private-user-images.githubusercontent.com/34395426/537234725-c12b9389-886b-4103-bc42-ccf95a562160.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njg3NDg1MzgsIm5iZiI6MTc2ODc0ODIzOCwicGF0aCI6Ii8zNDM5NTQyNi81MzcyMzQ3MjUtYzEyYjkzODktODg2Yi00MTAzLWJjNDItY2NmOTVhNTYyMTYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTE4VDE0NTcxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM5ZjhlY2E1OTY4MmE4MzU1ODUyZDRlZjQ1ZTEwYzNiNTdlODU5ODI1Y2NkNWZhODBkYzI0ZGZhMTZjNTU5ODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.n6FXoV10j5GO4Ve_z6cPFcwBf-t60D9M6_mM5zKDMC8">
> > > > I made the suit button 1 pixel narrower so the text can be centered in it, I'm trying to figure out whether it looks noticeably bad.
> > > 
> > > 
> > > It's kind of noticeable yeah, but I think you can make the S 1px wider to compensate?
> > > We could also do this for the Discard button, I'm noticing now the text isn't centered either.
> > 
> > 
> > Okay, but I can't change just this "S", the font in the UI has to be consistent so I changed all of the 2x5 S's in the UI to 3x5 including "**S**KIP", "**S**ELECT", "**S**MALL BLIND", "BO**SS** BLIND", "DI**S**CARD".
> > I took the S from the [m3x6](https://managore.itch.io/m3x6) font which is the closest parallel to Balatro's [m6x11](https://managore.itch.io/m6x11) font at this size. We should probably make a standardized font for the UI based on it.
> > The thing is, now the "SELECT" text during blind select is not centered within the button. The SKIP button is centered now though... Maybe centering all text in the buttons is a little much to hope for
> > <img alt="image" width="480" height="320" src="https://private-user-images.githubusercontent.com/34395426/537299583-27051898-ec83-4343-82a8-1add47066b73.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njg3NTcxODIsIm5iZiI6MTc2ODc1Njg4MiwicGF0aCI6Ii8zNDM5NTQyNi81MzcyOTk1ODMtMjcwNTE4OTgtZWM4My00MzQzLTgyYTgtMWFkZDQ3MDY2YjczLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTE4VDE3MjEyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlmM2E1YjMwNmIzOTA2YWZiYTJlNWIyNDUyMjI0OTA4MWRmNGIwODE4ZjVjZjlkN2Q2ZGM1ZjhkNjY0MTM3NzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.qeHFVLFql9GL0ZOnuq_WTgTegBSErxtpZ_tE-3wZYQ0">
> > <img alt="image" width="480" height="320" src="https://private-user-images.githubusercontent.com/34395426/537299679-a097d4e4-c0e7-4d62-82de-607677c552d1.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njg3NTcxODIsIm5iZiI6MTc2ODc1Njg4MiwicGF0aCI6Ii8zNDM5NTQyNi81MzcyOTk2NzktYTA5N2Q0ZTQtYzBlNy00ZDYyLTgyZGUtNjA3Njc3YzU1MmQxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTE4VDE3MjEyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEyNmM4ZDFhY2E5MDM3OTg2NmQyNjA2OTJjZjQyOWQ3MDYwZmE1ZTA5MTJlY2EyM2M0Mjg5MThiOGI0MDg0NTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.n0nFD9wBHSFLUPox-amspWOGRmlVF9Y60olC6p_spiA">
> 
> Yeah keeping everything consistent and centered looks like a chore alright. For my Planet and Tarot cards I sacrificed consistency in favor of centering when could but I think the UI looks very nice the way it is now that you changed it.
> 
> We _could_ sacrifice the C of all the letters in Select and make it 3 pixels wide to center the word, I think that would look the less weird and there is no other C to compare it to in that menu, but doing it or not is your call I don't want to impose more work on you than I already did ^^'

That's an idea, but then it means changing the C in "DIS**C**ARD" which means it won't be centered there again then... there is no escape, I'll have to think about it a little more, maybe the whole font changing should be postponed to another PR.

MathisMartin31 commented:
Yeah you're right. As I said, it's your call, and I do tend to just think about new things to do and roll with it without stopping and asking myself if it's within the current scope (I actually do that kind of stuff at work lol, fun times)

MeirGavish commented:
I'm opening it up again. I reverted the font changes, I think it should be left to another issue/PR. I changed the button size to be consistent with "RANK" but now the text is not centered. I guess it's not so bad since the text is not centered on the "SKIP" button too and we can look at centering text when making the changes to the font.

<img width="480" height="320" alt="image" src="https://github.com/user-attachments/assets/921a733c-b659-4c15-a1e8-1a6593f506a2" />

<img width="480" height="320" alt="image" src="https://github.com/user-attachments/assets/1924433f-6b4a-4e09-a8a1-28ff62665e42" />

MathisMartin31 commented:
Looks good!
